### DISCUSSION TITLE
Questions about the signature generated by the script-tester using Polkadot example script and argument

### DISCUSSION THREAD

**Discussion by cyje66** [CoolBitX-Technology/coolwallet-signing-tools#41] [Q&A] [answered]
**Description**

I'm currently testing the example Polkadot script and argument given in the CoolWallet Script Tester. I fill in the script and argument, and the generated signature looks like this: 
`f42b6a1573c6d75234f9d13ca0e3926df38f6988eb8fbe30da7603b2e197ec9a43151ad4e15555108c0224bb08ef180cf4cc79154734498b41494e076179bc6add4f7515bbdf3cff672769451aa99761`

I'm just curious about what the signature means. And when I send the same script and argument again, it'll give me a different signature. Shouldn't it be constant? (like `00003044022037AC2FFEEA6AF1DC07793EDBAC2FF1B6DE1E2812AD78D4C0D3CB950C6913726F022064370FC4A38247BB531396AD3C866DDB5A5350150C542806D9C1994A76F86645` in the sdk). Or am I taking it the wrong way. Help me out here, please.

**Expected Behavior**

Shouldn't the signature be like this in the sdk?
[https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/master/packages/coin-dot/src/config/params.ts](https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/master/packages/coin-dot/src/config/params.ts)
![image](https://user-images.githubusercontent.com/23631550/151022217-851ae6c1-771e-441b-94a9-c1f61b10ba82.png)

**Screenshots or appendix**

The example script and argument that I use:
![image](https://user-images.githubusercontent.com/23631550/151016959-181d50a1-a167-4e19-b4c8-ff54dae3dfb0.png)
script: `03020E01C7070000000162CAA0C70002CC071000CAAC570002A2ACD70022FFF6CAACD7002CFFFBA2ACD70031FFFBA2ACD70036FFFBCAACC7003B04CAACC7003F04CAAC570043CAAC570063DC07C003444f54CC0FC00753533538505245CC0F1000CAAC5F00025AF09FC00FBAFCCE6C07230804DDE09700DAACD7C022FFF60AD207CC05065052455353425554546f4e`
argument: `15328000002C8000016280000000000000000000000005008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a480000000000BEEBCF08000000009999000000000000000000009a0200004d01000091b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c35c697847b25d385178aa150d29e5ce212339c5624183f74bdf45f4912c89749a`

The signature that I generate:
![image](https://user-images.githubusercontent.com/23631550/151020950-d009dbd4-868b-434c-9b01-e4abfe85b65d.png)

**Comment by hankliu418** [comment]
Would you please give us you CoolWallet Card version?
We should have this information to answer your question, Thanks.

**Reply by cyje66** [reply]
Hi @hankliu418 ,
Do you mean the card's SE version? I think it's 320.
It was originally a commercial card, and I tried to upgrade it to a testing card with a specific token that CoolBitX team gave me. But I'm not sure if the upgrade is successful because when I finish upgrading it, the panel shows "AGAIN".

**Reply by hankliu418** [reply]
Hi @cyje66 
Thanks for replying. The result you mentioned 『f42b......9761』 is encrypted signature, should decrypt after you click button on CoolWallet card. We found some issue in coin-tester, and we will resolve soon.

**Reply by cyje66** [reply]
Hi @hankliu418 
I'm not sure what you mean there. When I got the signature(`f42b......9761`) and I pressed the button on CoolWallet card, nothing happened. The signature still remained encrypted. Moreover, if I do it again, it'll give me a different signature. Shouldn't it be the same even though it's encrypted?
How do I decrypt the signature?
(p.s. I'm using the Signing tool's script-tester not the coin-tester in the sdk)

Same argument and script but it returns a different signature.
![image](https://user-images.githubusercontent.com/23631550/151282086-2472897f-1fe5-4fc6-97db-27276543f85a.png)

Many thanks.

**Comment by hankliu418** [comment, marked as answer]
Hi @cyje66 
We have updated coolwallet-signing-tools, please pull the latest code in master branch, thanks.
And ECDSA signature will change every time, even you sign the same message, please refer the link below:
https://learnmeabitcoin.com/beginners/digital_signatures_signing_verifying

**Comment by cyje66** [comment]
Hi @hankliu418 
Got it! Thanks a lot!
One more question, the signature that I produce in the signing tools should be used as a parameter in the sdk, right?
For example, the signature above should be used in the signature field in the sdk?
[https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/master/packages/coin-dot/src/config/params.ts](https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/master/packages/coin-dot/src/config/params.ts)
![image](https://user-images.githubusercontent.com/23631550/151496771-c47626c0-db12-4109-8a88-833a6bff94ac.png)

**Comment by hankliu418** [comment]
Hi @cyje66 
The signature in parameter is generated by CoolBitX for this script. In develop step, you can use 『FA0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000』instead first.

And the signature produce in the signing tools is only used for transaction.

**Reply by cyje66** [reply]
Hi @hankliu418 
Okey...
But how do I know if the signature produced by the signing tools is valid?
How to send the transaction and broadcast to the blockchain?

**Comment by hankliu418** [comment]
You can use any programming language to verify signature is valid or not, and find free api to post transaction to blockchain.
